dallinger/prolific.py

```diff
@@ -149,23 +206,78 @@ def draft_study(
     ) -> dict:
         """Create a draft Study on Prolific, and return its properties."""
 
+        config = get_config()
+        config.load()
+
+        project_name = config.get("prolific_project")
+        workspace_name = config.get("prolific_workspace")
+        use_prolific_default_project = True
+
+        try:
+            # Get the workspace ID.  If it's not in Prolific, the function will raise an exception and create the workspace.
+            try:
+                workspace_id = self._translate_workspace_name(workspace_name)
+                use_prolific_default_project = False
+
+            except ProlificServiceNoSuchWorkspace:
+                # If the workspace does not exist in the '/workspaces/' endpoint, the '/studies/' endpoint will be
```

Comment:
This comment needs updating

Reply:
Sorry for being unclear, I meant that

> the '/studies/' endpoint

should be

> the '/workspaces/' endpoint

Reply:
But isn't the original and current comment talking about the last line in this method:
`return self._req(method="POST", endpoint="/studies/", json=payload)`
?

Reply:
I think the code has been copy-pasted and not updated properly. I think the comment refers to the logic concerning `use_prolific_default_project`, which I believe was originally written for the try-catch block that begins

```py
project_id = self._translate_project_name(workspace_id, project_name
use_prolific_default_project = False
```

and has then been copied up to the try-catch block that begins 

```py
workspace_id = self._translate_workspace_name(workspace_name)
use_prolific_default_project = False
```

Looking at the code, I don't think the `use_prolific_default_project` logic makes sense at all. The variable is initialised as `True` but no matter which route you take through the try-catch blocks it is always set to `False` by the end of the method, so it might as well not be used at all.

Reply:
I removed those comments altogether and also  the `use_prolific_default_project` variable.

I then tested again successfully all four combinations, see matrix above. 

@pmcharrison This is ready for a hopefully final review.

Reply:
@fmhoeger thank you for this. I've looked through the discussion history and realised that the problem came down to [this message](https://github.com/Dallinger/Dallinger/pull/6788#issuecomment-2278618242) which outlines a logic that doesn't quite make sense (I think there was a copy-paste error). Unfortunately I must have not read it carefully enough because I said that the logic was OK.

The question is, are we happy with the behaviour that we now have implemented? I think it's not so good to create workspaces or projects with the title 'Dallinger workspace' or 'Dallinger project', I think most users will want to customise that, especially because you can't delete workspaces as you found earlier.

How about the following:

Remove the default values of `prolific_project` and `prolific_workspace` from config. If they aren't specified, throw an error to the user and say they need to specify these in `.dallingerconfig` or `config.txt`? And that those projects or workspaces will be created automatically if they don't exist already?

I think that should be easy to implement?

Sorry for the problems here!!

Reply:
I removed the both default values from `global_config_defaults.txt`. The error is now raised in the `launch` route as 
![image](https://github.com/user-attachments/assets/49edff79-bce5-4a39-b2f9-0baddae7201a)

Is this a good place to raise the error? Or should it rather go inside somewhere in the `ProlificRecruiter` context?